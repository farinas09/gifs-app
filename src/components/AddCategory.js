import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2){
            //Arrow function has access to current state
        setCategories( cats => [inputValue, ...cats]);
    }
    setinputValue('');
    }

    return (
            <form className="form-group" onSubmit={handleSubmit}>
            <input className="form-control"
                placeholder="Search GIFs"
                type="text"
                value= {inputValue}
                onChange={handleInputChange}
            />
            </form>
    )
}

//required function setCategories
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
