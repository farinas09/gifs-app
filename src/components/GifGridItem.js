import React from 'react'
import { PropTypes } from 'prop-types'

export const GifGridItem = ({id, title, url}) => {
    console.log(id, title, url)
    return (
        <div className="card animate__animated animate__fadeIn">
            <img className="card-img-top" src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes ={
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
