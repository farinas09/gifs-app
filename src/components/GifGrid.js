import React from 'react'
import { PropTypes } from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3 className="animate__fadeIn">{category}</h3>
        { loading && <div class="alert alert-primary" role="alert">Loading</div> }
        <div className="card-grid album py-5 bg-light animate__fadeIn">
                {
                    images.map( (img) => (
                    <GifGridItem
                        key={img.id}
                        {...img} />
                        )) 
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}