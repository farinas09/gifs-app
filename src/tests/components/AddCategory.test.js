import { shallow } from 'enzyme';
import React from "react"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en <AddCategory />', () =>{
    const setCategories = () => {

    }
    const wrapper = shallow(<AddCategory setCategories = {setCategories} />)

    test('should show the component successfully', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})