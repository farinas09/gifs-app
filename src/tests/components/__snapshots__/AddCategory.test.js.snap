// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> should show the component successfully 1`] = `
<form
  className="form-group"
  onSubmit={[Function]}
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search GIFs"
    type="text"
    value=""
  />
</form>
`;
